libcontainer/system: move userns utilities, remove `GetParentNSeuid`, `UIDMapInUserNS`

thaJeztah opened the pull request:
### libcontainer/system: remove unused `GetParentNSeuid()`

This function was added in https://github.com/opencontainers/runc/commit/f103de57ec89137d86b77584f676b760ec7c7ba4 (https://github.com/opencontainers/runc/pull/1688), but no longer used since https://github.com/opencontainers/runc/commit/06f789cf26774dd64cb2a9cc0b3c6a6ff832733b (https://github.com/opencontainers/runc/pull/1862) (v1.0.0-rc6)

It is no longer used by runc itself, and only used by a  [single external project][3]

[3]: https://grep.app/search?q=.GetParentNSeuid


### libcontainer/system: un-export UIDMapInUserNS()

`UIDMapInUserNS()` is not used anywhere, only internally. so un-export it.


### libcontainer/system: move userns utilities to separate package

Moving these utilities to a separate package, so that consumers of this package don't have to pull in the whole "system" package.

Looking at uses of these utilities (outside of runc itself);

`RunningInUserNS()` is used by [various external consumers][1],
so adding a "Deprecated" alias for this.

[1]: https://grep.app/search?current=2&q=.RunningInUserNS

thaJeztah commented:
Opening as draft; I'll create a PR in Moby to see if it has the desired result, then move it out of draft

AkihiroSuda commented on libcontainer/system/userns_deprecated.go:
```diff
@@ -0,0 +1,17 @@
+var (
+	// RunningInUserNS detects whether we are currently running in a user namespace.
+	// Originally copied from github.com/lxc/lxd/shared/util.go
+	// Deprecated: use github.com/opencontainers/runc/libcontainer/userns.RunningInUserNS
```
We haven't been guaranteeing Go compatibility of libcontainer, so I think we can just drop these deprecated functions, but not a strong opinion.

thaJeztah replied:
Perhaps keep only "RunninginUserNS" (the "popular" one) and drop the others

thaJeztah commented:
I removed those functions that we don't use; reordered the commits to first remove these parts before moving the code. Moved this out of draft.

I also have a branch to isolate the userns package from the user package, as we only use the `user` package for a single function, and we likely don't need that; https://github.com/opencontainers/runc/pull/2868

thaJeztah commented:
@kolyshkin ptal 🤗

thaJeztah commented:
@kolyshkin PTAL (I'll rebase https://github.com/opencontainers/runc/pull/2868 after this)

kolyshkin reviewed (approved):
LGTM

kolyshkin commented:
close/reopen to kick ci

thaJeztah commented:
looks like ci/centos7 doesn't start; let me try another rebase to force CI to run

thaJeztah commented:
close/reopen to trigger fedora CI

thaJeztah commented:
rebased again to kick centos7

thaJeztah commented:
All green now 🎉  @kolyshkin @AkihiroSuda ptal

thaJeztah commented:
@kolyshkin ptal 🤗

kolyshkin reviewed (approved):
LGTM

thaJeztah commented:
Thanks! I rebased https://github.com/opencontainers/runc/pull/2868 and moved it out of draft
